src/libdredd/include_private/include/libdredd/mutate_visitor.h

```diff
@@ -46,23 +50,47 @@ class MutateVisitor : public clang::RecursiveASTVisitor<MutateVisitor> {
   bool TraverseConstantArrayTypeLoc(
       clang::ConstantArrayTypeLoc constant_array_type_loc);
 
-  // Overridden to avoid mutating variable array size expressions in C++
-  // (because lambdas cannot appear in such expressions).
+  // Overridden to avoid mutating array sizes that are derived from template
+  // parameters, because after template instantiation these lead to either
+  // constant or variable-sized arrays, neither of which can be mutated in C++.
```

Comment:
@JamesLee-Jones looks like the comment for this method and the next need to be swapped.